src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/ResourceTypeIdParameterQueryGenerator.cs

```diff
@@ -22,5 +23,22 @@ public override SearchParameterQueryGeneratorContext VisitString(StringExpressio
 
             return VisitSimpleBinary(BinaryOperator.Equal, context, VLatest.Resource.ResourceTypeId, expression.ComponentIndex, resourceTypeId);
         }
+
+        public override SearchParameterQueryGeneratorContext VisitIn(InExpression expression, SearchParameterQueryGeneratorContext context)
+        {
+            List<object> resolvedResourceTypeIds = new List<object>(capacity: expression.Values.Count);
+
+            foreach (string resourceType in expression.Values)
+            {
+                if (context.Model.TryGetResourceTypeId(resourceType, out short resourceTypeId))
+                {
+                    resolvedResourceTypeIds.Add(resourceTypeId);
+                }
+
+                // FHIBF: What do with invalid values? If customer pushes an invalid resource type, what should we do in such case?
```

Comment:
These have likely been "validated" several times by now with the `IsKnownResource` code. If we get to this point perhaps we should throw?